ENIGMAsystem/SHELL/Collision_Systems/General/collisions_general.cpp

```diff
@@ -0,0 +1,64 @@
+/** Copyright (C) 2021 Nabeel Danish  <nabeelben@gmail.com>
+*** This file is part of the ENIGMA Development Environment
+**/
+
+#include "collisions_general.h"
+
+namespace enigma
+{
+    // -------------------------------------------------------------------------
+    // Function to compute the bbox values used in many collision detection 
+    // functions
+    //
+    // Args: 
+    //      left, top, right, bottom    -- int values specifying the box, send 
+    //                                     them as reference since they get 
+    //                                     updated and returned 
+    //      inst                        -- instance for which to compute the 
+    //                                     bbox values
+    // -------------------------------------------------------------------------
+    void get_bbox_border(int &left, int &top, int &right, int &bottom, 
+        const enigma::object_collisions* inst, double x, double y) 
+    {
+        // Compute the bbox from the polygon 
+        // TODO (Nabeel) : Merge this IF condition when the polygon collision system 
+        //                  is merged
+        if (inst->polygon_index != -1)
+        {
+            // Using parameterized points if sent
+            double x1, y1;
+            if (x == -1 && y == -1)
+            {
+                x1 = inst->x;
+                y1 = inst->y;
+            } else {
+                x1 = x;
+                y1 = y;
+            }
+
+            // Fetching transformed points
+            std::vector<glm::vec2> points = enigma::polygons.get(inst->polygon_index).getPoints();
+            glm::vec2 pivot = enigma::polygons.get(inst->polygon_index).computeCenter();
+            enigma::transformPoints(points, 
+                                    x1, y1, 
+                                    inst->polygon_angle, pivot,
+                                    inst->polygon_xscale, inst->polygon_yscale);
+            
+            // Computing bbox
+            enigma::BoundingBox box = enigma::computeBBOXFromPoints(points);
+            left = box.left();
+            top = box.top();
+            right = box.right();
+            bottom = box.bottom();
+        }
+        // If the polygon is not availble, the bbox is computed from the polygon
+        else if (inst->sprite_index != -1)
+        {
+            const enigma::BoundingBox &box = inst->$bbox_relative();
+            const double x = inst->x, y = inst->y,
```

Comment:
I can change this to auto for the code consistency, however, there are places where we need double-precision, especially when working with transformations and concavity. I am not sure if the change will propagate in the codebase, but this can be refactored.

Reply:
Also because we do have `cs_scalar` if people want to change it. Those fields may need to be changed later.

Although Nabeel makes a great point about affine transformations, perhaps we could make another scalar for transforms.